os/drivers/audio/ndp120_voice.c

```diff
@@ -505,8 +509,20 @@ static int ndp120_enqueuebuffer(FAR struct audio_lowerhalf_s *dev, FAR struct ap
 			break;
 		}
 	}
+
+	if (ret != OK) {
```

Comment:
hmm I think below logic is required.
```
static void *ndp120_app_device_health_check(void)
{
	if (ndp_120_died) {
		dev->alive = false;
		reset_reload;
	}
	...
}

void reset_reload()
{
	do_reset_reload;
	dev->alive = true;
}

int ndp120_extract_audio(struct ndp120_dev_s *dev, struct ap_buffer_s *apb)
{
	if (!dev->alive) {
		return SYNTIANT_NDP_ERROR_UNINIT;
	}
	...
}

static int ndp120_enqueuebuffer(FAR struct audio_lowerhalf_s *dev, FAR struct ap_buffer_s *apb)
{
	...
	int ret = ndp120_extract_audio(priv, apb);
	/* Regardless result, dequee from pendq */
		for (tmp = (sq_entry_t *)sq_peek(&priv->pendq); tmp; tmp = sq_next(tmp)) {
		if (tmp == (sq_entry_t *)apb) {
			sq_rem(tmp, &priv->pendq);
			audvdbg("found the apb to remove 0x%x\n", tmp);
			break;
		}
	}
	
	if (ret == SYNTIANT_NDP_ERROR_UNINIT) {
		priv->dev.upper(priv->dev.priv, AUDIO_CALLBACK_UNREACHABLE, NULL, OK);
	}
	...
}
```

Reply:
Yes, thank you. This makes more sense for our case. I have updated it.